tt_metal/impl/dispatch/kernels/cq_dispatch.cpp

```diff
@@ -798,5 +810,13 @@ void kernel_main() {
                                                         wr_block_idx);
     noc_async_write_barrier();
 
+    if (is_h_variant && !is_d_variant) {
+        // Set upstream semaphore MSB to signal completion and path teardown
+        // in case dispatch_h is connected to a depacketizing stage.
+        // This should be replaced with a signal similar to what packetized components
```

Comment:
add a TODO on this line so we can grep for future work easily.  I try to use TODO for optional things we could/should do and XXXX for things that are broken or must dos, fwiw